m365-developer-proxy/Program.cs

```diff
@@ -17,4 +17,41 @@
 
 rootCommand.Handler = proxyHost.GetCommandHandler(pluginEvents, loaderResults.UrlsToWatch, logger);
 
-return await rootCommand.InvokeAsync(args);
\ No newline at end of file
+// store the global options that are created automatically for us
+string[] globalOptions = { "--version", "-?", "-h", "--help" };
+
+// filter args to retrieve short (-n) and long (--name) option aliases
+var incomingOptions = args.Where(arg => arg.StartsWith("-") || arg.StartsWith("--")).ToArray();
+
+// remove the global options from the incoming options
+incomingOptions = incomingOptions.Except(globalOptions).ToArray();
+
+// compare the incoming options against the root command options
+foreach (var option in rootCommand.Options)
+{
+    // get the option aliases
+    var aliases = option.Aliases.ToArray();
+
+    // iterate over aliases
+    foreach (string alias in aliases)
+    {
+        // if the alias is present
+        if (incomingOptions.Contains(alias))
+        {
+            // remove the option from the incoming options
+            incomingOptions = incomingOptions.Where(val => val != alias).ToArray();
+        }
+    }
+}
+
+// list the remaining incoming options as unknown in the output
+if (incomingOptions.Length > 0)
+{
+    logger.LogInfo($"Unknown option(s): {string.Join(" ", incomingOptions)}");
```

Comment:
These statements seem more like messages than actual command output. As such, I think we should output them to stderr rather than stdout. We can do it by using LogError instead. It will mean though that they'd be colored red. It could be useful to bring the attention to it (after all, something went wrong), but if it's too excessive, then we should consider introducing an extra logging method on `ILogger` which uses default printing style but sends the output to stderr.

Reply:
Done.